Form guidelines: Labels (#366)
nl-design-system/documentatie: docs/richtlijnen/formulieren/_label-accessible-name.md

Doen: Gekoppeld label met `for` en `id`. Het `<label>`-element heeft een `for` attribuut (of `htmlFor` in React) dat verwijst naar het `id` van het bijbehorende formulierveld.

Zo werkt dit bijvoorbeeld met een `<select>`:

```html
<label for="kleur">Wat is je lievelingskleur</label>
<select id="kleur" name="lievelingskleur">
  <option>Rood</option>
  <option>Blauw</option>
  <option>Wit</option>
</select>
```

Niet doen: Een `<div>` in plaats van een `<label>` gebruiken. De `for` en `id` koppeling werkt alleen met een `<label>`-element.

```html
<!-- Dit is foute code, niet kopiëren -->
<div for="kleur">Wat is je lievelingskleur</div>
<input type="text" id="kleur" name="lievelingskleur" />
```

Niet doen: Label zonder gekoppelde input. Door het `<label>`-element te gebruiken zonder `for` mis je de koppeling tussen het label en de input voor mensen die de visuele context niet waarnemen.

```html
<!-- Dit is foute code, niet kopiëren -->
<label>Wat is je lievelingskleur</label>
<select name="lievelingskleur">
  <option>Rood</option>
  <option>Blauw</option>
  <option>Wit</option>
</select>
```

> Het geven van een goede toegankelijke naam aan formuliervelden is verplicht volgens de volgende WCAG-succescriteria:

Author: dit kan nog iets concreter (en verwijzen naar 22 docs want die hebben de nieuwste CSS (long story)), zie suggesties hieronder

> - [2.4.6: Koppen en labels (niveau AA)](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels)

Author: ```suggestion
- [2.4.6: Koppen en labels (niveau AA)](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels): zegt dat een label het onderwerp of doel moet beschrijven

> - [3.3.2: Labels of instructies (niveau A)](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions)

Author: ```suggestion
- [3.3.2: Labels of instructies (niveau A)](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions): zegt dat formuliervelden een label of instructie moeten hebben (doel van die eis: gebruikers weten wat ze moeten invullen)
```

> - [4.1.2: Naam, rol, waarde (niveau A)](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)

Author: ```suggestion
- [4.1.2: Naam, rol, waarde (niveau A)](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html): zegt dat elementen (waaronder formuliervelden) een toegankelijke naam moeten hebben
```
